### ISSUE TITLE
Store GDEF in-UFO instead of in features.fea

### ISSUE THREAD

**Issue by madig** [googlefonts/googlesans#138]
Storing GDEF information in features.fea is possible and done today but is also very much annoying. One, it needs separate maintenance and two, it causes https://github.com/googlefonts/googlesans/issues/112.

The UFO spec now allows to generate a complete GDEF at compile time:

1. LigatureCaretByPos can be generated from https://github.com/unified-font-object/ufo-spec/blob/gh-pages/versions/ufo3/glyphs/glif.md#ligature-carets
2. GlyphClassDef can be generated from https://unifiedfontobject.org/versions/ufo3/glyphs/glif/#publicopentypecategory

So, _someone_ needs to sit down and write a GDEF writer for ufo2ft. Since it runs after all (anchor propagation) filters, we might not even need to store any categories. GDEF generation is already done by code in glyphsLib, so that can be used as a start.

**Comment by chrissimpkins** [comment]
> So, someone needs to sit down and write a GDEF writer for ufo2ft.

Has this been opened on the ufo2ft repo?

**Comment by madig** [comment]
https://github.com/googlefonts/ufo2ft/issues/456

**Comment by chrissimpkins** [comment]
> [googlefonts/ufo2ft#456](https://github.com/googlefonts/ufo2ft/issues/456)

Blocked on https://github.com/googlefonts/ufo2ft/issues/456

**Comment by madig** [comment]
This is relevant for https://github.com/googlefonts/googlesans/issues/148#issuecomment-777432018, as the desired category can be stored in the fonts instead of in external scripts.

**Comment by chrissimpkins** [comment]
It looks like Denis pushed updates in https://github.com/googlefonts/ufo2ft/pull/480 related to this issue.  Are we still blocked here @madig?  Something that we still want to do?

**Comment by madig** [comment]
Everything should work now once we put the relevant info into lib.plist. I need to do that at some point.
